pandas/core/algorithms.py

```diff
@@ -263,7 +263,7 @@ def _reconstruct_data(
     return values
 
 
-def _ensure_arraylike(values):
+def _ensure_arraylike(values) -> ArrayLike:
```

Comment:
is_array_like is True for Series and Index so the return type should probably be AnyArrayLike